modules/exploits/linux/http/panos_auth_rce.rb

```diff
@@ -0,0 +1,164 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::CmdStager
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  require 'ipaddr'
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Palo Alto Networks Authenticated Remote Code Execution',
+        'Description' => %q{
+          An OS Command Injection vulnerability in the PAN-OS management interface that allows authenticated
+          administrators to execute arbitrary OS commands with root privileges.
+          This issue impacts PAN-OS versions < 10.0.1, < 9.1.4 and < 9.0.10
+        },
+        'Author' => [
+          'Mikhail Klyuchnikov', # Vulnerability discovery
+          'Nikita Abramov', # Vulnerability discovery
+          'UnD3sc0n0c1d0',                          # Exploit
+          'jheysel-r7'                              # msf module
+        ],
+        'References' => [
+          ['CVE', '2020-2038'],
+          ['URL', 'https://swarm.ptsecurity.com/swarm-of-palo-alto-pan-os-vulnerabilities/'],
+          ['URL', 'https://security.paloaltonetworks.com/CVE-2020-2038'],
+          ['URL', 'https://github.com/und3sc0n0c1d0/CVE-2020-2038'] # Exploit
+        ],
+        'DisclosureDate' => '2020-09-09',
+        'License' => MSF_LICENSE,
+        'Platform' => 'linux',
+        'Arch' => ARCH_CMD,
+        'Privileged' => true,
+        'Targets' => [
+          [
+            'Linux x86',
+            {
+              'Platform' => 'linux',
+              'Arch' => ARCH_X86,
+              'CmdStagerFlavor' => %i[echo printf]
+            }
+          ],
+          [
+            'Linux x86_64',
+            {
+              'Platform' => 'linux',
+              'Arch' => ARCH_X64,
+              'CmdStagerFlavor' => %i[echo printf]
+            }
+          ]
+        ],
```

Comment:
You can combine the architectures for the Linux platform here into one. This will allow the user to select either x64 or x86 payloads when using this target. You can see an example of multiple architectures being specified in a target in the [psexec](https://github.com/rapid7/metasploit-framework/blob/master/modules/exploits/windows/smb/psexec.rb#L66) module.

```suggestion
        'Platform' => 'linux',
        'Privileged' => true,
        'Targets' => [
          [
            'Linux',
            {
              'Platform' => 'linux',
              'Arch' => [ARCH_X86, ARCH_X64],
              'CmdStagerFlavor' => %i[echo printf]
            }
          ]
```

Is it possible to also add an `ARCH_CMD` target? There doesn't appear to be any character or length restrictions on the payload.